src/LineBot/Events/LineEventCollection.cs

```diff
@@ -25,7 +24,7 @@ internal sealed class LineEventCollection : ILineEvents
         [JsonProperty("destination")]
         public string Destination { get; set; }
 
-        [JsonProperty("events")]
+        [JsonProperty("events", ItemConverterType = typeof(ConcreteConverter<ILineEvent, LineEvent>))]
```

Comment:
Wonder if you need the converter. Can you change it to LineEvent here and do an explicit implementation of the interface for the property?

Reply:
I think having a converter is our best bet as if the interface is implemented explicitly there will be a lot changes that must be made and then the Property will only be accessible from the interface and not from the `LineEventCollection` itself, like `(this as ILineEvents).Events`

Reply:
But the interface is the only thing that is visible to the outside world so that should not be an issue. And in the unit tests you could use `events.Count()` instead of `events.Events.Count()` so you don't need to do the typecast.

Reply:
See when newtonsoft sees the LineEventCollection it sees a destination and a list of interfaces and it doesn't know anything about LineEvent class so when it tries to deserialize the converter is the only think which tells newtonsoft as to how to deserialize. Furthermore, changing it to List<LineEvent> is also not possible because of the accessibility problems, where LineEvent is internal but the IlineEvents is public. So I think we should keep the converter, as it can also help later.

Reply:
You should change the type to `IEnumerable` instead of `List` in the `ILineEvents` interface. We don't want a mutable object in the public interface.

And then you don't need the converter because you can do this:

```C#
        [JsonProperty("events")]
        public List<LineEvent> Events { get; set; }

        IEnumerable<ILineEvent> ILineEvents.Events => Events;
```